include/styleUtil.h

```diff
@@ -0,0 +1,847 @@
+}
+
+/*
+ *  calculate a proper width for the TLegend object that can cover the entry with longest label (excluding the header)
+ */
+double calcTLegendWidth(TLegend* legend, double offset, double ratio, double threshold) {
+    
+    TIter iter(legend->GetListOfPrimitives());
+    TLegendEntry* entry;
+    double w = 0;
+    while (( entry = (TLegendEntry*)iter.Next() )) {
+        std::string label = entry->GetLabel();
+        if (legend->GetHeader() != NULL && label == legend->GetHeader())  continue;    // in this case, assume that the entry is actually the header
+        
+        double c = 1;
+        if (label.length() > 50) c = 1.5;
+        else if (label.length() > 30) c = 1.6;
+        else if (label.length() > 15) c = 1.4 + (label.length()-15)* 0.2 / 15 ;
+        
+        double tmp = c*ratio*label.length() + offset;
+        if (tmp < threshold) tmp = threshold;
+        if (tmp > w) w = tmp;
+    }
+    
+    return w;
+}
+
+void setCanvasTLatex(TCanvas* c, float px, float py, std::vector<std::string> lines, float pyOffset)
+{
+    c->cd();
+    
+    float pyNew = py;
+    TLatex* latex;
+    for (std::vector<std::string>::const_iterator it = lines.begin(); it != lines.end(); ++it){
+        
+        latex = new TLatex(px, pyNew, (*it).c_str());
+        latex->SetTextFont(43);
+        latex->SetTextSize(20*0.8);
+        latex->SetNDC();
+        latex->Draw();
+        pyNew = pyNew - pyOffset;
+    }
+}
+
+void setCanvas_InvMass(TCanvas* c, float px, float py, const char* collision, const char* energy,
+                       const char* ptCut, const char* etaCut, const char* extra)
+{
+    c->cd();
+    
+    float pyNew = py;
+    TLatex *latexCMS = new TLatex(px,pyNew, collision);
+    latexCMS->SetTextFont(43);
+    latexCMS->SetTextSize(20);
+    latexCMS->SetNDC();
+    latexCMS->Draw();
+    
+    pyNew = pyNew-0.05;
+    TLatex *latexEnergy = new TLatex(px, pyNew, energy);
+    latexEnergy->SetTextFont(43);
+    latexEnergy->SetTextSize(20);
+    latexEnergy->SetNDC();
+    latexEnergy->Draw();
+    
+    pyNew = pyNew-0.05;
+    TLatex *latexPt = new TLatex(px, pyNew, ptCut);
+    latexPt->SetTextFont(43);
+    latexPt->SetTextSize(20);
+    latexPt->SetNDC();
+    latexPt->Draw();
+    
+    pyNew = pyNew-0.06;
+    TLatex *latexEta = new TLatex(px, pyNew, etaCut);
+    latexEta->SetTextFont(43);
+    latexEta->SetTextSize(20);
+    latexEta->SetNDC();
+    latexEta->Draw();
+    
+    std::string str = extra;
+    if (str != "")
+    {
+        pyNew = pyNew-0.05;
+        TLatex *latexExtra = new TLatex(px, pyNew, extra);
+        latexExtra->SetTextFont(43);
+        latexExtra->SetTextSize(20);
+        latexExtra->SetNDC();
+        latexExtra->Draw();
+    }
+}
+
+void setCanvas_InvMass_pp(TCanvas* c, float px, float py, const char* ptCut, const char* etaCut, const char* extra)
+{
+    setCanvas_InvMass(c, px, py, "CMS preliminary pp", "#sqrt{s} = 5.02 TeV", ptCut, etaCut, extra);
```

Comment:
This script seems to have some extraneous CMS stuff. Are you using this file anywhere?

Reply:
I use it for some of the histogram formatting. I'm deleting all the CMS specific code.